pygac/klm_reader.py

```diff
@@ -587,29 +587,40 @@ def read(self, filename):
                 The scanlines
 
         """
-        super(KLMReader, self).read(filename=filename)
```

Comment:
The original code of the abstract method only sets the attribute `reader.filename` which I think should not contain the *.gz* extension to be consistent with not compressed input files in the hdf5 output. Since the extension is split away in the `Reader._open` method anyway, I think it would be good to set the attribute `reader.filename` there depending on the input file extension. This change makes the code of the abstract method obsolete and in order to keep up with the standard practice, I replaced the abstract method `Reader.read` with raising a `NotImplementedError`.

Reply:
Sounds very good.